indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/SeekableStreamIndexTaskRunner.java

```diff
@@ -1887,10 +1887,13 @@ public Response pauseHTTP(
   @VisibleForTesting
   public Response pause() throws InterruptedException
   {
-    if (!(status == Status.PAUSED || status == Status.READING)) {
+    // Read the volatile status into a variable so that its value does not change while the condition is evaluated
+    final Status currentStatus = status;
+    if (!(currentStatus == Status.PAUSED || currentStatus == Status.READING)) {
```

Comment:
TBH: 1890 does not add a lot of value :). 

I think the valid state transitions should reside int the status class. That way we have one central way of controlling the transitions instead of having the logic spread all over the place. 

Can be a follow up PR as well .

Reply:
Yeah, we need to revisit this class in earnest.

Reply:
> TBH: 1890 does not add a lot of value

Do you mean the comment?